File: spring_java_orm_sample/src/main/java/com/example/demo/config/LoginRateLimitProperties.java
package com.example.demo.config;

import jakarta.validation.constraints.Min;
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;
import org.springframework.validation.annotation.Validated;

import java.time.ZoneId;

@Component
@ConfigurationProperties(prefix = "app.auth.login-rate-limit")
@Validated
@Getter
@Setter
public class LoginRateLimitProperties {
    private boolean enabled = true;

    @Min(1)
    private int dailyLimit = 10;

    private ZoneId zoneId = ZoneId.of("Asia/Tokyo");
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/config/LoginRateLimitProperties.kt
package com.example.demo.config

import jakarta.validation.constraints.Min
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component
import org.springframework.validation.annotation.Validated
import java.time.ZoneId

@Component
@ConfigurationProperties(prefix = "app.auth.login-rate-limit")
@Validated
class LoginRateLimitProperties {
    var enabled: Boolean = true

    @field:Min(1)
    var dailyLimit: Int = 10

    var zoneId: ZoneId = ZoneId.of("Asia/Tokyo")

    fun isEnabled(): Boolean = enabled
}
